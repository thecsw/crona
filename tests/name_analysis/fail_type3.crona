b : bool;
arr : byte array[10];
main : int (size : int, str : string) {
    i : int;
    i = 0;
    while (i < size) {
        b = b && arr[i] == str[i];
        if (b) {
            b : int;
            b = arr[i];
            write b;
        }
        else {
            write b;
        }
    }
    return i;
}