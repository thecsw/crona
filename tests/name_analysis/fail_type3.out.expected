b(bool):bool;
arr(byte array [10]):byte array[10];
main(int,byte array [0]->int):int(size(int):int, str(byte array [0]):byte array[0]){
	i(int):int;
	i(int) = 0;
	while ((i(int) < size(int))){
		b(bool) = (b(bool) && (arr(byte array [10])[i(int)] == str(byte array [0])[i(int)]));
		if (b(bool)){
			b(int):int;
			b(int) = arr(byte array [10])[i(int)];
			write b(int);
		} else {
			write b(bool);
		}
	}
	return i(int);
}